src/data_collector/indicator_pipeline/indicator_pipeline.py

```
@@ -234,12 +362,43 @@
+                                    "feature_value": feature_value,
+                                    "quality_score": quality_score_val,
+                                }
+                            )
+
+                    saved_count = bulk_upsert_technical_features(
+                        rows, page_size=1000, overwrite=config.overwrite_existing
+                    )
+                    logger.info(
+                        f"Saved {saved_count} feature records to database for {ticker}"
+                    )
+                except Exception as e:
```

Ruff (0.12.2)

Do not catch blind exception: `Exception`

(BLE001)